vignettes/z_code_for_article_in_musik_und_internet.Rmd

```diff
@@ -0,0 +1,119 @@
+library(psychTestR)
+
+devtools::install_github('fmhoeger/psyquest')
+library(psyquest)
+
+library(shiny)
+ 
+# Codeblöcke
+save_time_started <- code_block(function(state, ...) { # Speichern der Startzeit 
+  set_local(                                           # lokal für diesen Teilnehmer
+    key = "time_started",
+    value = Sys.time(),
+    state = state  # interne Statusvariable von psychTestR
+  )
+})
+ 
+compute_time_taken <- code_block(function(state, ...) {        # Berechnung und Ausgabe          
+  time_taken <- Sys.time() - get_global("time_started", state) # der Bearbeitungsdauer
+  msg <- paste0(
+    "So lange hast du den Fragebogen bearbeitet: ",
+    format(time_taken, digits = 3)
+  )
+  shiny::showNotification(msg)  
+})
+ 
+timeline_abb_1 <- join( # Timeline aus Abbildung 1
+  one_button_page( # Willkommensseite
+    button_text = "Weiter",
+      shiny::div(
+        p("Willkommen zum", tags$strong("Musikfragebogen!")),
+        p(tags$em("Wir wollen heute mehr über Musik in deinem Leben erfahren."))
+      )
+  ),
+  text_input_page( # erste Frage (Instrument)
+    label = "Instrument",
+    prompt = "Welches Instrument spielst du?",
+    button_text = "Weiter",
+    on_complete = function(answer, state, ...) {
+      set_global(
+        key = "instrument",
+        value = answer,# answer enthält immer die letzte Antwort
+        state = state  # interne Statusvariable von psychTestR 
+      )
+  }),
+  reactive_page(function(state, answer, ...) { # zweite Frage (Dauer)
+    msg <- sprintf(
+        "Wie lange spielst du schon %s?",
+        get_local("instrument", state) # holt den aktuellen Wert aus state
+    )
+    NAFC_page(
+        label = "time_instrument",
+        msg,
+        c(
+          "Ich habe erst angefangen.",
+          "ein Monat bis ein halbes Jahr",
+          "0,5 bis 1 Jahr"
+        )
+    )
+  })
+)
+ 
+# verschiedene Pages zu einer Timeline durch join() bündeln
+timeline <- join(
+  new_timeline(                                  # Timeline für die Startzeit
+    save_time_started,                           # Timelines brauchen ein Dictionary, das die
+    dict = psyquest::psyquest_dict               # internationalisierten Sprachbausteine 
+                                                 # enthält, hier nur ein Dummy.
+  ),
+  timeline_abb_1,                                # Timeline aus Abbildung 1
+  GMS(language = "de"),                          # Goldsmiths Musical Sophistication Index
+  compute_time_taken,                            # Berechnung und Ausgabe Bearbeitungsdauer
+  elt_save_results_to_disk(complete = TRUE),     # Speicherung der Ergebnisse
+  final_page("Vielen Dank!")                     # Abschlussseite
+)
+ 
+# fertigen Fragebogen öffnen
+questionnaire <- make_test(
+  timeline,
+  opt = test_options(
+    title = "Musikfragebogen",
+    admin_password = "abc",
+    languages = "de"
+  )
+)
+ 
+shiny::runApp(questionnaire)
+
+# Zugang zu den Ergebnissen
```

Comment:
Likewise here, it's a bit confusing to include data import code in the app.R itself. Maybe make separate sections of the Rmd file that clearly delineate that these bits of code belong to different steps? i.e. finish the code block, add some text, then start a new code block?

Reply:
Right! I added a separate code bloxk. (In the article we didn't speak about file structure and the app.R file, that's why I put it all together, but I'm happier now with the solution you've suggested. Thank you!)